gyp/pylib/gyp/easy_xml.py

```diff
@@ -121,7 +123,10 @@ def WriteXmlIfChanged(content, path, encoding="utf-8", pretty=False, win32=False
 
     default_encoding = locale.getdefaultlocale()[1]
     if default_encoding and default_encoding.upper() != encoding.upper():
-        xml_string = xml_string.encode(encoding)
+        if win32 and sys.version_info < (3, 7):
```

Comment:
@owl-from-hogvarts This causes failure on windows python 3.6. Can you take a look.

Reply:
also cc @cclauss

Reply:
Hey! I had a breif view on that. My guess is that in line:
```diff
+ xml_string = xml_string.decode("cp1251").encode(encoding)
```
`decode("cp1251")` should not be hardcoded but set to `default_encoding` like:
```diff
- xml_string = xml_string.decode("cp1251").encode(encoding)
+ xml_string = xml_string.decode(default_encoding).encode(encoding)
```

Reply:
@gengjiawen ☝️

Reply:
I am not a Windows guy but it would be helpful if someone can add tests that fail in our GitHub Actions.

Reply:
@owl-from-hogvarts From the test, still failed. Should we just revert this part ?

Reply:
Sad. Then we need somehow get encoding of `xml_string` to properly decode it. In general, where did `xml_string` come from? I mean from file or form stdin or where?

Reply:
@gengjiawen ☝️